compendium/DeclarativeServices/src/manager/states/CCActiveState.cpp

```diff
@@ -23,10 +23,24 @@
 #include "CCActiveState.hpp"
 #include "CCUnsatisfiedReferenceState.hpp"
 #include "../ComponentConfigurationImpl.hpp"
+#include "cppmicroservices/SharedLibraryException.h"
 
 namespace cppmicroservices {
 namespace scrimpl {
 
+class LatchScopeGuard {
+public:
+  LatchScopeGuard(std::function<void()> cleanupFcn) : _cleanupFcn(std::move(cleanupFcn))
+  {
+    
+  }
+  ~LatchScopeGuard() {
+    _cleanupFcn();
```

Comment:
Make sure this destructor cannot throw. `latch.CountDown()` can throw so this destructor can throw. I suggest a `try/catch` and log the exception in the `catch` if its safe to do so (i.e. if `Log` doesn't throw).

Reply:
will try/catch around latch.CountDown(). No mention of any throws from Log (both observing the code and looking at the OSGi spec).